kurbo/src/line.rs

```diff
@@ -158,19 +158,27 @@ impl ParamCurveArea for Line {
 }
 
 impl ParamCurveNearest for Line {
+    #[inline]
     fn nearest(&self, p: Point, _accuracy: f64) -> Nearest {
         let d = self.p1 - self.p0;
-        let dotp = d.dot(p - self.p0);
-        let d_squared = d.dot(d);
-        let (t, distance_sq) = if dotp <= 0.0 {
-            (0.0, (p - self.p0).hypot2())
-        } else if dotp >= d_squared {
-            (1.0, (p - self.p1).hypot2())
-        } else {
-            let t = dotp / d_squared;
-            let dist = (p - self.eval(t)).hypot2();
-            (t, dist)
-        };
+        let v = p - self.p0;
+
+        // Calculate projection parameter `t` of the point onto s(t), with s(t) the line segment
+        // such that s(t) = (1-t) * p0 + t * p1.
+        //
+        // Note this will be inf when the segment has 0 length; see the clamping below.
+        let t = d.dot(v) / d.hypot2();
+
+        // Clamp the parameter to be on the line segment. This results in `t==0` if `t==inf` above.
+        #[expect(
+            clippy::manual_clamp,
+            reason = "`clamp` uses slightly more instructions than chained `max` and `min` on x86 and aarch64"
```

Comment:
aarch64: https://godbolt.org/z/Mv4Y19jrM
x86: https://godbolt.org/z/WW19hMd6v

Perhaps we should disable this lint in `kurbo`.

Reply:
Seems that way! (As a separate note, we actually use some of the `NaN` semantics of the chained-min-max in the analytical pixel coverage calculation in the sparse strips work, so this lint can be a bit noisy.)